libs/slam/async_slam/async_slam.cpp

```
@@ -220,13 +220,12 @@ void AsyncSlam::TrackResult(const FrameId frameId, const int64_t timestamp_ns,
 }
 
 Isometry3T AsyncSlam::GetSlamPose() const {
-  std::lock_guard slam_guard(slam_mutex_);
-  const auto may_be_tip = tail_.GetTip();
+  const auto may_be_tip = tail_.GetTip();  // thread-safe
   if (!may_be_tip) {
     return Isometry3T::Identity();
   }
   const Isometry3T& tail_tip = may_be_tip->second;
-  Isometry3T slam_pose = tail_tip * track_data_.from_keyframe;
+  Isometry3T slam_pose = tail_tip * track_data_.from_keyframe;  // track_data_ touched only in main thread
```

Clang (14.0.6)

[warning] variable 'slam_pose' is not initialized

(cppcoreguidelines-init-variables)